specs/separate_configuration_files_per_component.adoc

```diff
@@ -0,0 +1,157 @@
+= Specification for segregating configurations per module wise
+
+This specification describes instead of writing all configuration details into
+file, Store everything in etcd and configure all tendrl modules via api request.
+
+== Problem description
+
+Currently, All the tendrl modules are configured from same file. Also, while
```

Comment:
Yeah, all the tendrl modules are configured via `tendrl.conf`, but that was your design choice.